homeassistant/components/bluetooth/models.py

```diff
@@ -0,0 +1,107 @@
+"""Models for bluetooth."""
+from __future__ import annotations
+
+import asyncio
+import logging
+from typing import Any, Final
+
+from bleak import BleakScanner
+from bleak.backends.device import BLEDevice
+from bleak.backends.scanner import AdvertisementData, AdvertisementDataCallback
+
+from homeassistant.core import CALLBACK_TYPE, callback as hass_callback
+
+_LOGGER = logging.getLogger(__name__)
+
+FILTER_UUIDS: Final = "UUIDs"
+
+HA_BLEAK_SCANNER: HaBleakScanner | None = None
+
+
+class HaBleakScanner(BleakScanner):
+    """BleakScanner that cannot be stopped."""
+
+    _callbacks: list[AdvertisementDataCallback] = []
```

Comment:
This is safe as HaBleakScanner is a singleton?

Reply:
Yes 👍